src/lexstate.c

```diff
@@ -112,17 +115,60 @@ int rbs_token_bytes(rbs_token_t tok) {
 }
 
 unsigned int rbs_peek(rbs_lexer_t *lexer) {
-    if (lexer->current.char_pos == lexer->end_pos) {
-        lexer->last_char = '\0';
-        return 0;
+    return lexer->current_code_point;
```

Comment:
The `rbs_peek` simply returns the `current_code_point` that was read in `rbs_next_char`. We can skip calling expensive `rbs_utf8_string_to_codepoint` function.